Raise minimum CMake version to 3.10

cjmayo opened the pull request:
For compatibility with CMake 4.0, which also removes CMP0043 OLD - there are no uses of `COMPILE_DEFINITIONS_<CONFIG>`.

---

I've used this on Linux with CMake 4.0.2.

SuperSandro2000 commented:
Parts of this where already merged with https://github.com/intel/intel-graphics-compiler/commit/4f0123a7d67fb716b647f0ba5c1ab550abf2f97d . Can you rebase the PR?

@obrotowy

cjmayo committed:
Raise minimum CMake version to 3.10

For compatibility with CMake 4.0, which also removes CMP0043 OLD - there
are no uses of COMPILE_DEFINITIONS_<CONFIG>.

Signed-off-by: Chris Mayo <aklhfex@gmail.com>
